src/actions/index.js

```diff
@@ -0,0 +1,275 @@
+/* eslint-disable comma-dangle */
+import axios from 'axios';
+
+export const ROOT_URL = 'https://open-stata.herokuapp.com/api';
+
+export const ActionTypes = {
+  AUTH_USER: 'AUTH_USER',
+  DEAUTH_USER: 'DEAUTH_USER',
+  AUTH_ERROR: 'AUTH_ERROR',
+  GET_DOFILES: 'GET_DOFILES',
+  CREATE_DOFILE: 'CREATE_DOFILE',
+  GET_SINGLE_DOFILE: 'GET_SINGLE_DOFILE',
+  SAVE_DOFILE: 'SAVE_DOFILE',
+  DELETE_DOFILE: 'DELETE_DOFILE',
+  GET_LOGFILES: 'GET_LOGFILES',
+  GET_SINGLE_LOGFILE: 'GET_SINGLE_LOGFILE',
+  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
+  GET_DATA: 'GET_DATA',
+  GET_TUTORIALS: 'GET_TUTORIALS',
+};
+
+export function signinUser(user, history, onError) {
+  return (dispatch) => {
+    axios
+      .post(`${ROOT_URL}/signin`, user)
+      .then((response) => {
+        localStorage.setItem('token', response.data.token);
+        dispatch({ type: ActionTypes.AUTH_USER, payload: response.data });
+        history.push('/home');
+      })
+      .catch((error) => {
+        console.error(error);
+        // dispatch(`Sign In Failed: ${error.response.data}`);
+        onError();
+      });
+  };
+}
+
+export function signupUser({ email, password, username }, history, onError) {
+  return (dispatch) => {
+    axios
+      .post(`${ROOT_URL}/signup`, { email, password, username })
+      .then((response) => {
+        localStorage.setItem('token', response.data.token);
+        dispatch({ type: ActionTypes.AUTH_USER, payload: response.data });
+        history.push('/home');
+      })
+      .catch((error) => {
+        dispatch(`Sign Up Failed: ${error.response.data}`);
+        onError();
+      });
+  };
+}
+
+export function signoutUser(history) {
+  return (dispatch) => {
+    localStorage.removeItem('token');
+    dispatch({ type: ActionTypes.DEAUTH_USER });
+    history.push('/');
+  };
+}
+
+export function authError(error) {
+  return {
+    type: ActionTypes.AUTH_ERROR,
+    message: error,
+  };
+}
+
+export const getDoFiles = (setInitialized) => (dispatch) => {
+  axios
+    .get(`${ROOT_URL}/dofiles`, {
+      headers: { authorization: localStorage.getItem('token') },
+    })
+    .then((res) => {
+      dispatch({
+        type: ActionTypes.GET_DOFILES,
+        payload: res.data,
+      });
+      if (setInitialized) setInitialized(true);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
+
+export const getTutorialFiles = (setNext, setInitialized) => (dispatch) => {
+  axios
+    .get(`${ROOT_URL}/tutorials`, {
+      headers: { authorization: localStorage.getItem('token') },
+    })
+    .then((res) => {
+      dispatch({
+        type: ActionTypes.GET_TUTORIALS,
+        payload: res.data,
+      });
+      if (setNext) setNext(setInitialized);
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
have some error component show some of these, otherwise this is just an invisible fail